### PR TITLE
Bug 1349642, added a new section about restrictions on route host and…

### PR DESCRIPTION
**Pull request by ahardin-rh**
… certificates for users' routes

https://bugzilla.redhat.com/show_bug.cgi?id=1349642

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@adellape @bfallonf @tnguyen-rh @tpoitras please peer review, noting that the newly added section applies to OpenShift Online and Dev Preview :bow:

@Miciah PTAL

File: dev_guide/routes.adoc [hunk 1]
```diff
@@ -133,10 +134,33 @@ custom certificates. Any non-SNI traffic received on port 443 is handled with
...
 as well as
 xref:../architecture/core_concepts/routes.adoc#path-based-routes[path-based
 routing] are available in the
 xref:../architecture/core_concepts/routes.adoc#architecture-core-concepts-routes[Architecture section].
+
+ifdef::openshift-online[]
+[[custom-route-and-hosts-and-certificates-restrictions]]
+== Restrictions
+
+Currently, custom route hosts are not permitted. Instead, the following host
```
Comment (Miciah) [inline dev_guide/routes.adoc]: There is some nuance beyond what the qualifier "currently" conveys; namely, this restriction will be lifted for accounts with paid plans, but not for unpaid accounts (such as on Developer Preview).  For users with unpaid accounts, the word "currently" may be understood to mean, incorrectly, that the restriction is temporary.

**Comment by Miciah** [comment]
One minor concern around wording, otherwise looks fine.  Thanks!

File: dev_guide/routes.adoc (hunk 1 above)
Comment (tpoitras) [inline dev_guide/routes.adoc]: I see what @Miciah is saying here. So something like "Custom route hosts are only available for users with paid accounts", except there's bound to be a better way to indicate a paid account without saying "paid accounts". In the older version of OpenShift Online, we had the "Bronze Plan" and "Silver Plan" for the paid tiers. Do we know yet what the paid tiers of the new Online will be called?

File: dev_guide/routes.adoc [hunk 2]
```diff
@@ -133,10 +134,33 @@ custom certificates. Any non-SNI traffic received on port 443 is handled with
...
+template is enforced on all user routes:
+
+----
+<route-name>-<namespace>.44fs.preview.openshiftapps.com
+----
+
+Also, custom certificates are not permitted. Only unencrypted routes, edge
+routes using the default certificate, and passthrough routes work. Edge routes
+with custom certificates and re-encrypt routes (which necessarily have custom
+certificates) will not work.
```
Comment (tpoitras) [inline dev_guide/routes.adoc]: s/will/do

File: dev_guide/routes.adoc [hunk 3]
```diff
@@ -133,10 +134,33 @@ custom certificates. Any non-SNI traffic received on port 443 is handled with
...
+
+These restrictions are enforced in the router. Inspecting the
+route in the console or using `oc get routes` will display the host and
```
Comment (tpoitras) [inline dev_guide/routes.adoc]: s/will display/displays

File: dev_guide/routes.adoc [hunk 4]
```diff
@@ -133,10 +134,33 @@ custom certificates. Any non-SNI traffic received on port 443 is handled with
...
+certificates specified by the user. However, any custom host will not be
```
Comment (tpoitras) [inline dev_guide/routes.adoc]: s/any custom host will not/custom hosts are not

File: dev_guide/routes.adoc [hunk 5]
```diff
@@ -133,10 +134,33 @@ custom certificates. Any non-SNI traffic received on port 443 is handled with
...
+respected, and a route with custom certificates will not work.
```
Comment (tpoitras) [inline dev_guide/routes.adoc]: s/will not/cannot ?

There's a lot of future tense here.

File: dev_guide/routes.adoc [hunk 6]
```diff
@@ -133,10 +134,33 @@ custom certificates. Any non-SNI traffic received on port 443 is handled with
...
+endif::[]
```
Comment (bfallonf) [inline dev_guide/routes.adoc]: @ahardin-rh A lot of "will"-ness. Can we change the tense to non-future?

"custom hosts are not respected"
"routes with custom certificates do not work"

**Comment by bfallonf** [comment]
@ahardin-rh 1 minor nit... LG otherwise.

**Comment by ahardin-rh** [comment]
@tpoitras @bfallonf Thanks for the peer review! Updates complete.

@Miciah @tpoitras For now, I removed "currently"; I discussed this with @adellape and we previously referenced "tiers" of service and will probably return to that language for our purposes here.

**Comment by Miciah** [comment]
Looks good!

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../dev_guide/routes.adoc#dev-guide-routes[Routes]
|Added a new section about xref:../dev_guide/routes.adoc#custom-route-and-hosts-and-certificates-restrictions[restrictions on route host and certificates for users' routes].
%

**Pull request merged**